### PR TITLE
fix ssrc messed up when generating trackinfo

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

File: pkg/rtc/mediatrackreceiver.go [hunk 1]
```diff
@@ -568,9 +568,15 @@ func (t *MediaTrackReceiver) TrackInfo(generateLayer bool) *livekit.TrackInfo {
 					ci.Layers = append(ci.Layers, proto.Clone(layer).(*livekit.VideoLayer))
 
 					// if origin layer has ssrc, don't override it
-					if layerIdx < len(originLayers) && originLayers[layerIdx].Ssrc != 0 {
-						ci.Layers[layerIdx].Ssrc = originLayers[layerIdx].Ssrc
```
Comment (cnderrauber) [inline pkg/rtc/mediatrackreceiver.go:572]: `getVideoLayersLocked` returned value don't have same order as protocol layers, cause ssrc messed up

File: pkg/rtc/mediatrackreceiver.go [hunk 2]
```diff
@@ -568,9 +568,15 @@ func (t *MediaTrackReceiver) TrackInfo(generateLayer bool) *livekit.TrackInfo {
...
-					} else if int(layer.Quality) < len(receiver.layerSSRCs) {
+					ssrcFound := false
```
Comment (boks1971) [inline pkg/rtc/mediatrackreceiver.go:571]: @cnderrauber I was just about to open a PR to do the same thing :-). Will review this and abort my attempt.

File: pkg/rtc/mediatrackreceiver.go [hunk 3]
```diff
@@ -568,9 +568,15 @@ func (t *MediaTrackReceiver) TrackInfo(generateLayer bool) *livekit.TrackInfo {
...
+					for _, l := range originLayers {
+						if l.Quality == ci.Layers[layerIdx].Quality {
```
Comment (boks1971) [inline pkg/rtc/mediatrackreceiver.go:573]: The older code also had a check for `l.Ssrc != 0`. My diff looks like
```
+                                       originHasSsrc := false
+                                       for _, originLayer := range originLayers {
+                                               if originLayer.Quality != layer.Quality {
+                                                       continue
+                                               }
+
+                                               if originLayer.Ssrc != 0 {
+                                                       // if origin layer has ssrc, don't override it
+                                                       ci.Layers[layerIdx].Ssrc = originLayer.Ssrc
+                                                       originHasSsrc = true
+                                                       break
+                                               }
+                                       }
+
+                                       if !originHasSsrc && int(layer.Quality) < len(receiver.layerSSRCs) {
                                                ci.Layers[layerIdx].Ssrc = receiver.layerSSRCs[layer.Quality]
                                        }
```

**Comment by boks1971** [comment]
Although not necessary, I added this sort in `getVideoLayersLocked` just as way to return consistently ordered list
```
+       sort.Slice(layers, func(i, j int) bool {
+               return layers[i].Quality < layers[j].Quality
+       })
+
```

File: pkg/rtc/mediatrackreceiver.go (hunk 3 above)
Comment (boks1971) [reply pkg/rtc/mediatrackreceiver.go:573]: Actually, the break should be outside the `if` which sets the `ssrc`. As long as quality match is found, whether setting `ssrc` or not, it should break.

**Comment by cnderrauber** [comment]
@boks1971 updated the zero check

**Review by boks1971 — APPROVED** [review]

**Pull request merged**